packages/driver-interface/botRoutes.js

```diff
@@ -1,66 +1,68 @@
 const botServices = require('./services/bot')
 const messaging = require('./services/messaging')
-const { getVehicle } = require('./services/cache')
-
+const cache = require('./services/cache')
 const {
   open,
   exchanges: { INCOMING_BOOKING_UPDATES },
 } = require('./adapters/amqp')
 
-function onArrive(msg) {
+function onArrived(msg) {
   const telegramId = msg.update.callback_query.from.id
-  const vehicleId = msg.metadata.getVehicleIdFromTelegramId(telegramId)
+  const vehicleId = cache.getVehicleIdFromTelegramId(telegramId)
 
-  return botServices.handlePickupInstruction(vehicleId, telegramId)
+  return botServices.handleDriverArrivedToPickupOrDeliveryPosition(
+    vehicleId,
+    telegramId
+  )
 }
 
 function onPickup(msg) {
   const telegramId = msg.update.callback_query.from.id
-  const vehicleId = msg.metadata.getVehicleIdFromTelegramId(telegramId)
+  const vehicleId = cache.getVehicleIdFromTelegramId(telegramId)
 
   const callbackPayload = JSON.parse(msg.update.callback_query.data)
 
-  open
+  return open
     .then((conn) => conn.createChannel())
     .then((ch) => {
       ch.assertExchange(INCOMING_BOOKING_UPDATES, 'topic', {
         durable: false,
       }).then(() => {
         const { id } = callbackPayload
 
-        ch.publish(
+        return ch.publish(
           INCOMING_BOOKING_UPDATES,
           'picked_up',
           Buffer.from(JSON.stringify({ id, status: 'picked_up' }))
         )
       })
     })
-
-  return botServices.handleOnArrive(vehicleId, telegramId)
+    .then(() => botServices.handleNextDriverInstruction(vehicleId, telegramId))
+    .catch(console.warn)
 }
 
 function onDelivered(msg) {
   const telegramId = msg.update.callback_query.from.id
-  const vehicleId = msg.metadata.getVehicleIdFromTelegramId(telegramId)
+  const vehicleId = cache.getVehicleIdFromTelegramId(telegramId)
 
   const callbackPayload = JSON.parse(msg.update.callback_query.data)
-  open
+
+  return open
     .then((conn) => conn.createChannel())
     .then((ch) => {
       ch.assertExchange(INCOMING_BOOKING_UPDATES, 'topic', {
         durable: false,
       }).then(() => {
         const { id } = callbackPayload
-        ch.publish(
+        return ch.publish(
           INCOMING_BOOKING_UPDATES,
           'delivered',
           Buffer.from(JSON.stringify({ id, status: 'delivered' }))
         )
       })
     })
+    .then(() => botServices.handleNextDriverInstruction(vehicleId, telegramId))
```

Comment:
This changes the logic a bit, how come you moved this?

Reply:
Just a mistake